File: concepts/1_arrays_and_strings/unique_string/unique.php
<?php

echo is_unique( 'abcdefghijklmnopqrstuvwxyz' ) . "\n"; // true
echo is_unique( 'abcdefghijklmnopqrstuvwxyzz' ) . "\n"; // false
echo is_unique( 'aa' ) . "\n"; // false
echo is_unique( 'aba' ) . "\n"; // false
echo is_unique( 'aaaaaaaaaaaaaaaaaaaaaaaaaaa' ) . "\n"; // false
echo is_unique( '  ' ) . "\n"; // false
echo is_unique( ' ' ) . "\n"; // true

function is_unique( $string ) {
	$strlen = strlen( $string ); // calculate once instead of twice

	if ( 128 < $strlen ) { // TODO understand the difference between ascii and unicode
		return 'false';
	}

	$alf = array();
	for ( $i = 0; $i <= $strlen; $i++ ) { // not foreach
		$char = substr( $string, $i, 1 );

		if ( ! $alf[ $char ] ) { // true (already set)
			$alf[ $char ] = 'true';
		} else { // false
			return 'false';
		}
	}

	return 'true';
}

File: concepts/1_arrays_and_strings/unique_string/unique.py
def is_unique( str ):
	strlen = len( str );

	if strlen > 128:
		return;

	checker = []

	for letter in str:
		checker.append( letter );

	if len(checker) > len(set(checker)):
		return 'false'

	return 'true'

print is_unique("abcdefghijklmnopqrstuvwxyz"); # true
print is_unique("abcdefghijklmnopqrstuvwxyzz"); # false
print is_unique("aa"); # false
print is_unique("aba"); # false
print is_unique("aaaaaaaaaaaaaaaaaaaaaaaaaaa"); # false
print is_unique("  "); # false
print is_unique(" "); # true
